packages/firestore/src/lite-api/aggregate.ts

```diff
@@ -67,43 +68,42 @@ export class AggregateQuerySnapshot {
 /**
  * Creates an `AggregateQuery` counting the number of documents matching this query.
  *
- * @return An `AggregateQuery` object that can be used to count the number of documents in
+ * @returns An `AggregateQuery` object that can be used to count the number of documents in
  * the result set of this query.
  */
 export function countQuery(query: Query<unknown>): AggregateQuery {
   return new AggregateQuery(query);
 }
 
 export function getAggregateFromServerDirect(
-  aggregateQuery: AggregateQuery
+  query: AggregateQuery
 ): Promise<AggregateQuerySnapshot> {
-  const firestore = cast(aggregateQuery.query.firestore, Firestore);
+  const firestore = cast(query.query.firestore, Firestore);
   const datastore = getDatastore(firestore);
   const userDataWriter = new LiteUserDataWriter(firestore);
 
-  return invokeRunAggregationQueryRpc(datastore, aggregateQuery).then(
-    result => {
-      hardAssert(
-        result[0] !== undefined,
-        'Aggregation fields are missing from result.'
-      );
+  return invokeRunAggregationQueryRpc(datastore, query).then(result => {
+    hardAssert(
+      result[0] !== undefined,
+      'Aggregation fields are missing from result.'
+    );
 
-      const countField = (result[0] as any).count_alias;
-      hardAssert(
-        countField !== undefined,
-        'Count field is missing from result.'
-      );
+    const counts = Object.entries(result[0])
+      .filter(([key, value]) => key === 'count_alias')
+      .map(([key, value]) => userDataWriter.convertValue(value as Value));
 
-      const countAggregateResult = userDataWriter.convertValue(countField);
-      hardAssert(
-        typeof countAggregateResult === 'number',
-        'Count aggeragte field is not a number: ' + countAggregateResult
-      );
-      return Promise.resolve(
-        new AggregateQuerySnapshot(aggregateQuery, countAggregateResult)
-      );
-    }
-  );
+    const count = counts[0];
+    hardAssert(
```

Comment:
Can we get rid of this specific hardAssert for `count !== undefined`? It seems to be covered by the hardAssert below for `typeof count === 'number'`.